retrace.in

```diff
@@ -0,0 +1,112 @@
+#!/bin/sh
+
+readonly __progname="$(basename $0)"
+
+errx() {
+	echo -e "${__progname}: $@" >&2
+	exit 1
+}
+
+usage() {
+	echo "usage: ${__progname} [-f configuration file location] <executable>"
+	exit 1
+}
+
+get_lib_file() {
+    local fname="$1"
+
+    # check for the local file
+    local path="$fname"
+    if [ -f "$path" ]; then
+        output="$path"
+        return 0
+    fi
+
+    # check in autotools generated dir
+    local path=".libs/$fname"
+    if [ -f "$path" ]; then
+        output="$path"
+        return 0
+    fi
+
+    # check in autotools generated dir
+    local path="../.libs/$fname"
+    if [ -f "$path" ]; then
+        output="$path"
+        return 0
+    fi
+
+    # check in the PREFIX path
+    local path="@libdir@/$fname"
+    if [ -f "$path" ]; then
+        output="$path"
+        return 0
+    fi
+
+    # check default library path
+    local path="/usr/lib/$fname"
+    if [ -f "$path" ]; then
+        output="$path"
+        return 0
+    fi
+
+    # check default library path
+    local path="/usr/lib64/$fname"
+    if [ -f "$path" ]; then
+        output="$path"
+        return 0
+    fi
+
+    return 1
+}
+
+
+main() {
+	[ $# -lt 1  ] && \
```

Comment:
why does it have such crazy formatting? Tab is used here, but not down below. Starange line splitting, etc.